HW15/HW15.cpp

```diff
@@ -0,0 +1,73 @@
+#include <iostream>
+#include <string>
+#include <algorithm>
+
+struct Student {
+    std::string name;
+    int marks[4];
+
+
+    Student(const std::string& n, int m1, int m2, int m3, int m4) {
+        name = n;
+        marks[0] = m1;
+        marks[1] = m2;
+        marks[2] = m3;
+        marks[3] = m4;
+    }
+
+
+    double averageMarks() const {
+        int sum = 0;
+        for (int i = 0; i < 4; ++i) {
+            sum += marks[i];
+        }
+        return static_cast<double>(sum) / 4.0;
+    }
+};
+
+
+bool compareStudents(const Student& s1, const Student& s2) {
+    return s1.averageMarks() > s2.averageMarks();
+}
+
+
+Student* getTopStudent(Student* students, int count) {
+    if (count <= 0) {
+        return nullptr;
+    }
+    std::sort(students, students + count, compareStudents);
```

Comment:
please, make this task without sorting. it's faster to make this task without it